### ISSUE TITLE
No titlebar on GNOME 40 / Fedora 34 / Wayland

### ISSUE THREAD

**Issue by wjt** [flathub/org.fedoraproject.MediaWriter#14]
When I install and run 4.2.2 from Flathub built from commit a62d887c on my Fedora 34 system using Wayland-flavoured GNOME, the app launches with no titlebar:

![Screenshot from 2021-08-17 10-58-02](https://user-images.githubusercontent.com/86760/129706046-37a53d1e-6050-4231-ad8c-b89cbe1296a1.png)

If I run it with `--nosocket=wayland` the titlebar is displayed:

```
$ flatpak run --nosocket=wayland org.fedoraproject.MediaWriter
C@4ms: Failed to create wl_display (No such file or directory)
I@4ms: Could not load the Qt platform plugin "wayland" in "" even though it was found.
```

If I install and run `mediawriter-4.2.2-2.fc34.x86_64` it runs as an X11 app:

```
$ mediawriter
W@0ms: Could not find the Qt platform plugin "wayland" in ""
```

To confirm this I tried to run it with `DISPLAY` unset, and it failed to start:

```
$ DISPLAY= mediawriter 
W@1ms: Could not find the Qt platform plugin "wayland" in ""
W@7ms: could not connect to display 
I@7ms: Could not load the Qt platform plugin "xcb" in "" even though it was found.
F@7ms: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
```

If the app does not work correctly under Wayland, the Flatpak should not have `--socket=wayland`.

**Comment by grulja** [comment]
It definitely works under Wayland, the app doesn't do or doesn't need anything special to run on Wayland. Does any other Qt app suffer from the same problem? Can you try to update KDE runtime?

**Comment by wjt** [comment]
> Does any other Qt app suffer from the same problem?

I looked at the other Qt apps using the same runtime on this Fedora system.

- io.bit3.WhatsAppQT: OK       
- net.sonic_pi.SonicPi: doesn't have `--socket=wayland` permission but if I add it it still works fine
- org.kde.ktuberling: OK
- org.keepassxc.KeePassXC: OK

(org.kde.gcompris is fullscreen but is too large to fit on the screen. Unrelated issue/regression!)


> Can you try to update KDE runtime?

```
$ flatpak update org.kde.Platform/x86_64/5.15
Looking for updates…
Nothing to do.
$ flatpak info org.kde.Platform/x86_64/5.15

KDE Application Platform - Shared libraries used by KDE applications

          ID: org.kde.Platform
         Ref: runtime/org.kde.Platform/x86_64/5.15
        Arch: x86_64
      Branch: 5.15
     License: GPL-2.0+
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 1.0 GB

      Commit: 1d61fde96c732a939f32d225138df433d9d19ef2162ae2d5dabf958225996485
      Parent: 212d267860a3001081816007b074dd57993335aec23485c082d40034c6a5b857
     Subject: build of org.kde.Sdk, Mon 12 Jul 2021 12:34:23 AM UTC (4059f6be7b66ce27501484314de2b998cc92fcab)
        Date: 2021-07-12 06:44:23 +0000
```

I reproduced the same issue on an Endless OS machine (manually switched to Wayland, GNOME 3.38 based) and in a GNOME OS (41 nightly) VM.

**Comment by grulja** [comment]
Does it make any difference when you install `org.kde.PlatformTheme.QGnomePlatform` and `org.kde.WaylandDecoration.QGnomePlatform-decoration`?

**Comment by wjt** [comment]
Skipping: org.kde.PlatformTheme.QGnomePlatform/x86_64/5.15 is already installed
Skipping: org.kde.WaylandDecoration.QGnomePlatform-decoration/x86_64/5.15 is already installed

**Comment by grulja** [comment]
Should be fixed with https://github.com/flathub/org.fedoraproject.MediaWriter/commit/1aa43e3579580cebb74b3cb2282d4f866f0d62f2. Can you try to update? I could verify this in Weston.

**Comment by grulja** [comment]
There was miscompatibility between Adwaita-qt used by FMW and by Adwaita-qt used by QGnomePlatform.

**Comment by hfiguiere** [comment]
> net.sonic_pi.SonicPi: doesn't have `--socket=wayland` permission but if I add it it still works fine

no it doesn't. There are many bugs.

**Comment by wjt** [comment]
> Should be fixed with [1aa43e3](https://github.com/flathub/org.fedoraproject.MediaWriter/commit/1aa43e3579580cebb74b3cb2282d4f866f0d62f2). Can you try to update? I could verify this in Weston.

Yes, this fixed it for me. Thanks!

> > net.sonic_pi.SonicPi: doesn't have `--socket=wayland` permission but if I add it it still works fine
> 
> no it doesn't. There are many bugs.

"Works" = "has a titlebar" :) I didn't test any further than that.
